kubeflow/optimizer/backends/base.py

```diff
@@ -44,3 +50,17 @@ def get_job(self, name: str) -> OptimizationJob:
     @abc.abstractmethod
     def delete_job(self, name: str):
         raise NotImplementedError()
+
+    @abc.abstractmethod
+    def wait_for_job_status(
+        self,
+        name: str,
+        status: set[str] = {constants.OPTIMIZATION_JOB_COMPLETE},
+        timeout: int = 3600,
```

Comment:
Let's use same timeout as for Trainer for now?
```suggestion
        timeout: int = 600,
```

Reply:
I increased it since the default num_trials is 10, meaning optimization will take about 10 times longer than just training, right?

Reply:
I see, it should be fine in that case.